src/cpu/aarch64/cpu_isa_traits.hpp

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+* Copyright 2018-2020 Intel Corporation
+* Copyright 2020 FUJITSU LIMITED
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_CPU_ISA_TRAITS_HPP
+#define CPU_AARCH64_CPU_ISA_TRAITS_HPP
+
+#include <type_traits>
+
+#include "dnnl_types.h"
+
+#include "common/dnnl_thread.hpp"
+#include "common/utils.hpp"
+
+/* in order to make selinux happy memory that would be marked with X-bit should
+ * be obtained with mmap */
+#define XBYAK_USE_MMAP_ALLOCATOR
+
+#include "cpu/aarch64/xbyak_aarch64/xbyak_aarch64.h"
+#include "cpu/aarch64/xbyak_aarch64/xbyak_aarch64_util.h"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+enum cpu_isa_bit_t : unsigned {
+    asimd_bit = 1u << 0,
+    sve_128_bit = 1u << 1,
+    sve_256_bit = 1u << 2,
+    sve_384_bit = 1u << 3,
+    sve_512_bit = 1u << 4,
+};
+
+enum cpu_isa_t : unsigned {
+    isa_any = 0u,
+    asimd = asimd_bit,
+    sve_128 = sve_128_bit | asimd,
+    sve_256 = sve_256_bit | asimd,
+    sve_384 = sve_384_bit | asimd,
+    sve_512 = sve_512_bit | asimd,
+    isa_all = ~0u,
+};
+
+const char *get_isa_info();
+
+cpu_isa_t DNNL_API get_max_cpu_isa_mask(bool soft = false);
+status_t set_max_cpu_isa(dnnl_cpu_isa_t isa);
+dnnl_cpu_isa_t get_effective_cpu_isa();
+
+template <cpu_isa_t>
+struct cpu_isa_traits {}; /* ::vlen -> 32 (for avx2) */
+
+// pack struct so it can fit into a single 64-byte cache line
+#pragma pack(push, 1)
+struct palette_config_t {
+    uint8_t palette_id;
+    uint8_t startRow;
+    uint8_t reserved[14];
+    uint16_t cols[16];
+    uint8_t rows[16];
+};
+#pragma pack(pop)
```

Comment:
this is definitely irrelevant for Aarch64 (this is for Intel AMX)

Reply:
Code not related to AArch64 has been removed.
https://github.com/oneapi-src/oneDNN/pull/850/commits/8bb226766bdd94777223c5751032ca442cb4dbee